cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -271,13 +274,15 @@ public String toString() {
         private final boolean needsRestart;
         private final boolean needsReconfig;
         private final KafkaBrokerConfigurationDiff diff;
+        private final KafkaBrokerLoggingConfigurationDiff logDiff;
         private final Admin adminClient;
 
-        public RestartPlan(Admin adminClient, boolean needsRestart, boolean needsReconfig, KafkaBrokerConfigurationDiff diff) {
+        public RestartPlan(Admin adminClient, boolean needsRestart, boolean needsReconfig, KafkaBrokerConfigurationDiff diff, KafkaBrokerLoggingConfigurationDiff logDiff) {
```

Comment:
TBH, I'm not completely convinced it's worth adding the dynamic logging changes to the `KafkaRoller`. `KafkaRoller's` job is to safely orchestrate the restart or reconfiguration of some or all brokers in the cluster. Changing a broker's logging doesn't seem to me to be a very risky thing to do. It's not going to need any special pre- or post-conditions before/after happening, for example. I can see that from a certain PoV it makes sense to do it here, but the question is whether adding extra complexity to `KafkaRoller` to do it here really makes sense, compared with some other phase of the reconciliation which updates the logging after brokers have been restarted/reconfigured.